specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+* It isn't required that developers Register a Push Activator for Visual Toast operations. In the case of Visual Toasts, payloads are directed to the Shell and not to the App.
+
+## Handling Push ChannelFailures
+PushNotification channel request calls are expected to fail since they go over the wire. We don't want the developer to deal with the complexities of retryable failures and critical failures. So we expose in-progress states and internal error codes for developers to action on.
+
+For example, we have 2 different Progress states, one indicating basic progress and another indicating retry progress.
+```cpp
+if (args.Status() == ChannelStatus::InProgress)
+{
+    // This is basically a noop since it isn't really an error state
+    printf("The first channel request is still in progress! \n");
+}
+else if (args.Status() == ChannelStatus::InProgressRetry)
+{
+    LOG_HR_MSG(args.ExtendedError(), "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it.");
+}
+```
+
+Similarly, operations are expected to fail or succeed on completion.
+```cpp
+if (result.Status() == ChannelStatus::CompletedSuccess)
+{
+    auto channelUri = result.Channel().Uri();
+    auto channelExpiry = result.Channel().ExpirationTime();
+
+    // Persist the channelUri and Expiry in the App Service
+}
+else if (result.Status() == ChannelStatus::CompletedFailure)
+{
+    LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+}
+```
+
+## Manifest Registration
+For MSIX, the COM activator GUID and the exe path need to be registered in the manifest. The launch args would need to be pre set to a well-known string that defines Push Triggers. 
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="-ReunionPushServer">
+        <com:Class Id="{GUIDEntryPoint}" DisplayName="Reunion Push" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+## UWP InProc activator
+The UWP inproc activator flow differs from the Win32 inproc activator flow since there is technically no WinMain activation associated with UWPs. Instead the App will have to  override and implement OnBackgroundActivated which is tightly coupled to Windows.UI.Xaml workstreams.
+
+# API Details
+```c# (but really MIDL3)
+    namespace Microsoft.ProjectReunion
+    {
+        // The legacy ActivationKind does not support operations like Push. So we need to define an extension set
+        enum ActivationKindExtension
```

Comment:
This type should probably go into @aeloros 's API work, not here, per earlier discussion? This PR should just show the new "kind"

Reply:
Yes all references to ActivationKind will be removed and I will be deferring to @andreww-msft and @aeloros to come up with the API surface.